lib/kafka_ex/server.ex

```diff
@@ -285,7 +289,19 @@ defmodule KafkaEx.Server do
       # credo:disable-for-next-line Credo.Check.Refactor.CyclomaticComplexity
       def kafka_server_produce(produce_request, state) do
         correlation_id = state.correlation_id + 1
-        produce_request_data = Produce.create_request(correlation_id, @client_id, produce_request)
+        produce_request_data = try do
+          Produce.create_request(correlation_id, @client_id, produce_request)
+        rescue
```

Comment:
How are we arriving at this exception?
Should we have a base case for `Produce.create_request/3` which returns `:invalid_produce_request` or something like that?

Reply:
> How are we arriving at this exception?

For context, it's part of the changes in this commit: https://github.com/kafkaex/kafka_ex/pull/297/commits/254429e8e5143f270cd0768fe886436cca197278 .

I added this commit because I encountered a bug when I used `KafkaEx.produce` with messages that were simple maps rather than `%Message{}`. When I did that, KafkaEx would hang and the process would eat up all my system's memory very quickly. I rearranged the functions in `protocol/produce.ex` so that there would be no incorrect function match and added this exception to send back a nice error rather than just failing.

> Should we have a base case for Produce.create_request/3 which returns :invalid_produce_request or something like that?

Seems better. I'm not sure how to do it though, since the function clause error happens deeper in the stack, when `create_message_set_uncompressed` is called. Any hint ?

Reply:
I don't think I'm that concerned about it, feel free to ignore this comment.